packages/nextjs/src/config/webpack.ts

```diff
@@ -377,13 +382,27 @@ function checkWebpackPluginOverrides(
  *
  * @param entryPointName The name of the entry point in question
  * @param isServer Whether or not this function is being called in the context of a server build
+ * @param excludedServersideEntrypoints A list of excluded serverside entrypoints
  * @returns `true` if sentry code should be injected, and `false` otherwise
  */
-function shouldAddSentryToEntryPoint(entryPointName: string, isServer: boolean): boolean {
+function shouldAddSentryToEntryPoint(
+  entryPointName: string,
+  isServer: boolean,
+  excludedServersideEntrypoints: (string | RegExp)[],
+): boolean {
+  const isServerSideExcluded = excludedServersideEntrypoints.some(serverSideExclude => {
+    if (typeof serverSideExclude === 'string') {
+      return entryPointName === serverSideExclude;
+    } else {
+      return entryPointName.match(serverSideExclude);
+    }
+  });
+
   return (
-    entryPointName === 'pages/_app' ||
-    (entryPointName.includes('pages/api') && !entryPointName.includes('_middleware')) ||
-    (isServer && entryPointName === 'pages/_error')
+    (!isServer || !isServerSideExcluded) &&
+    (entryPointName === 'pages/_app' ||
+      (entryPointName.includes('pages/api') && !entryPointName.includes('_middleware')) ||
+      (isServer && entryPointName === 'pages/_error'))
```

Comment:
M: Given that we're checking `isServer` in multiple places, and given that our list of conditions is getting pretty unwieldy and hard to parse, what would you think of reorganizing this into a series of conditionals, kinda like

```js
if (entryPointName.includes('_middleware')) {
  return false;
}

if (isServer) {
  const isExcluded =  excludedServersideEntrypoints.some( ... );
  if (entryPointName === 'pages/_error' || isExcluded) {
    return false;
  }
}

if (etc) { ... }
```

I just think it would be a lot easier to grok. As a small bonus, that way we wouldn't have to do the `excludedServerSideEntryPoints` calculation if we're not building server files.

Reply:
Fully agree! Even thought about doing this myself when originally working on this PR. Changed in 63b713c1bc1669203ea3577e465fc6f7e1c73e82 - let me know if this is approximately what you wanted.

Plucking this apart though. I am surprised by the things we're including here. Do you happen to know the history of the following?:
- Why are we not including any non-API `pages`?
- Why serverside and clientside `_app` but only serverside `_error`?

Reply:
> * Why are we not including any non-API `pages`?

Because they all end up descendants of the `_app` component (put another way, the `_app` component is (almost) the root of the entire component tree), so as long as the SDK inits when `_app` runs, it's guaranteed to be init-ed on every page. (That has the bonus that on the client side, it means the code will only be included in someone's bundle once. Hopefully Webpack would fix that in any case, but injecting into only one spot makes it a non-issue.)

> * Why serverside and clientside `_app` but only serverside `_error`?

Similar reason on the client side - `_error` ends up being a descendent of `_app`. On the server side, it's possible for `_error` to run without `_app`, so we have to inject there, too.

Reply:
TIL: Six months ago middleware [moved from needing to be called `_middleware` to being called `middleware`](https://github.com/vercel/next.js/pull/36772). So ever since 12.1.7, that check hasn't even been checking the right thing. 😞

Reply:
> Because they all end up descendants of the `_app` component (put another way, the `_app` component is (almost) the root of the entire component tree), so as long as the SDK inits when `_app` runs, it's guaranteed to be init-ed on every page. (That has the bonus that on the client side, it means the code will only be included in someone's bundle once. Hopefully Webpack would fix that in any case, but injecting into only one spot makes it a non-issue.)

Potentially dumb question: Is it the case that we're not adding instrumentation if the project doesn't have an `_app` file?

Reply:
Nextjs will supply one if the user doesn't. I added a comment in my PR which builds off of this one, https://github.com/getsentry/sentry-javascript/pull/6206.